refactor fedora message sending into cosalib/fedora_messaging_request.py

dustymabe opened the pull request:
This PR re-factors out the code that sends requests to fedora
releng/infra via fedora messaging messages into a library. It can
then be called by the code in src/cmd-sign as well as new consumers
in the future.

Also implements request_id for fedmsg passing in cmd-sign.

dustymabe commented:
so I got this to work for [sending a message to staging](https://apps.stg.fedoraproject.org/datagrepper/id?id=2020-f775f5de-4fc9-4ee0-8e7b-ac1c2b8ea422&is_raw=true&size=extra-large) but it never responded. I'll have to investigate if staging robosig is in a good state or not.

darkmuggle commented on src/cosalib/fedora_messaging_request.py:
```diff
@@ -0,0 +1,172 @@
+    reactor.run(installSignalHandlers=False)  # pylint: disable=E1101
+
+
+if __name__ == '__main__':
```
Is this needed still?

dustymabe replied:
removed

dustymabe commented:
ok I pushed up an update:

- rebased on top of latest master
- fixed a bunch of flake8 issues
- ran python black on the two files I touched in order to format them (fixed a bunch of flake8 spacing errors)

dustymabe commented:
also figured out the issue with staging and successfully [sent a request](https://apps.stg.fedoraproject.org/datagrepper/id?id=2020-82a99357-2165-4f90-817e-2492994345c4&is_raw=true&size=extra-large) and got a [response](https://apps.stg.fedoraproject.org/datagrepper/id?id=2020-02b4ead9-20a4-449a-b5b9-d59009599be7&is_raw=true&size=extra-large)

jlebon commented on src/cosalib/fedora_messaging_request.py:
```diff
@@ -0,0 +1,165 @@
+
+def send_request_and_wait_for_response(config=None,
+                                       environment='prod',
+                                       request_type=None,
```
Feels like this one shouldn't have a default. E.g. maybe let's change the signature to `..._response(request_type, config=None, ...)`?

dustymabe replied:
should be fixed in the next upload

jlebon commented on src/cmd-sign:
```diff
@@ -23,68 +23,81 @@ import boto3
-    robosig.add_argument("--s3", metavar='<BUCKET>[/PREFIX]', required=True,
-                         help="bucket and prefix to S3 builds/ dir")
+    robosig = subparsers.add_parser(
+        "robosignatory", help="sign with " "RoboSignatory via fedora-messaging"
```
Looks like it missed merging the two strings here during the line split.

dustymabe replied:
will be fixed in the next upload

dustymabe commented on src/cosalib/fedora_messaging_request.py:
```diff
@@ -0,0 +1,168 @@
+    instance. For more details, see:
+
+    https://fedora-messaging.readthedocs.io/en/latest/quick-start.html
+"""
```
I tried to update the messages in both cmd-sign and here. Let me know what you thing after my next upload.

jlebon replied:
Yup, LGTM. :+1:

dustymabe commented on src/cmd-sign:
```diff
@@ -17,148 +17,141 @@ import subprocess
+from cosalib.fedora_messaging_request import send_request_and_wait_for_response
 
-gi.require_version('OSTree', '1.0')
+gi.require_version("OSTree", "1.0")
```
i just ran /usr/bin/black` from python3-black on it. This was mostly a lazy way to get rid of the spacing errors I was seeing from flake8. I can undo that change and fix the spacing changes manually if you prefer.

jlebon replied:
Not against using something like `black` on the repo, though it should be part of a larger discussion with the rest of the team I think.

> I can undo that change and fix the spacing changes manually if you prefer.

Sure, WFM!

dustymabe replied:
ok removed the `black` reformatting

dustymabe commented:
- removed `black` formatting
- rebased on top of latest master
- re-tested against robosignatory in staging

jlebon commented:
/lgtm
